blog/content/post/2025-07-18-securing-embedded-linux.md

```diff
@@ -0,0 +1,1190 @@
+  Installing STREAM zarhus-base-image-debug-genericx86-64.direct.p4.gz, 933239273 bytes
+(...)
+[INFO ] : SWUPDATE successful ! SWUPDATE successful !
+```
+
+After update, we can check that we booted from correct partitions:
+
+```text
+NAME               MAJ:MIN RM  SIZE RO TYPE  MOUNTPOINTS
+sda                  8:0    0 26.3G  0 disk
+|-sda1               8:1    0  128M  0 part
+|-sda2               8:2    0  128M  0 part  /boot
+|-sda3               8:3    0    3G  0 part
+| `-luks_rootfs_a  252:1    0    3G  0 crypt
+|-sda4               8:4    0    3G  0 part
+| `-luks_rootfs_b  252:0    0    3G  0 crypt /media/rfs/ro
+`-sda5               8:5    0   20G  0 part
+  `-luks_rwoverlay 252:2    0   20G  0 crypt /media/rfs/rw
+sr0                 11:0    1 1024M  0 rom
+```
+
+Or verify that BootOrder is correct:
+
+```text
+BootCurrent: 0003
+Timeout: 0 seconds
+BootOrder: 0003,0001,0000,0002,0004
+Boot0000* UiApp FvVol(7cb8bdc9-f8eb-4f34-aaea-3ee4af6516a1)/FvFile(462caa21-7614-4503-836e-8ab6f4662331)
+Boot0001* ZarhusOS A    HD(1,GPT,4e2f9fb1-985c-d28b-8092-25cc455a3e36,0x1000,0x40000)/\EFI\BOOT\bootx64.efi
+Boot0002* EFI Internal Shell    FvVol(7cb8bdc9-f8eb-4f34-aaea-3ee4af6516a1)/FvFile(7c04a583-9e3e-4f1c-ad65-e05268d0b4d1)
+Boot0003* ZarhusOS B    HD(2,GPT,fe12879d-0509-540c-6b36-e2f313253979,0x41000,0x40000)/\EFI\BOOT\bootx64.efi
+Boot0004* UEFI QEMU HARDDISK QM00001    PciRoot(0x0)/Pci(0x1f,0x2)/Sata(0,65535,0){auto_created_boot_option}
+```
+
+## What's next
+
+There are multiple possibilities, but next step would be to change encryption
+workflow a little:
+
+- encrypt with user supplied password
+- enroll TPM2 if system booted with SB enabled
+
+Current update implementation could probably be also improved, e.g. by creating
+installer or expanding first boot scripts we could create slot B during runtime
+which would result in much smaller images (half the size without compression).
+It should also be possible to use only single partition image in `.swu` update
+image and either keep any files that differ between slots as separate files in
+`.swu` archive or modify them in `otab-shell` script after inactive partition is
+updated.
+
+I'm also nearly finished with release CI for `meta-zarhus` which will allow us to
+create releases for multiple platforms with different features. I hope we
+will expand this CI in the future to include public cache upload to lower build
+time and possibly add more sophisticated tests like [DTS E2E
+tests](https://github.com/Dasharo/open-source-firmware-validation/blob/develop/dts/dts-e2e.robot).
+Those would be very helpful in detecting any problems or regressions. This is
+especially important as we have multiple features, and it is hard to test every
+feature and platform combination manually.
+
+If anyone is interested then I invite you to
+[Zarhus Developers
+Meetup](https://events.dasharo.com/event/6/zarhus-developers-meetup-2) where
+I'll also present what I've written here in more abbreviated form.
```

Comment:
I don't think this paragraph is a good pitch.

We could say that if anyone is interested in asking questions, they can join ZDM. I guess there will also be a demo on ZDM, so maybe it is worth mentioning that also?